src/freenet/support/CurrentTimeUTC.java

```diff
@@ -3,61 +3,49 @@
  * http://www.gnu.org/ for further details of the GPL. */
 package freenet.support;
 
-import java.util.Calendar;
+import java.time.OffsetDateTime;
+import java.time.ZoneOffset;
+import java.time.temporal.ChronoField;
 import java.util.Date;
-import java.util.GregorianCalendar;
-import java.util.TimeZone;
 
 
 /**
- * A wrapper class around a GregorianCalendar which always returns the current time.
- * This is useful for working around the pitfall of class Calendar: It only returns the current time when you first use a get*() function,
- * in any get*() calls after the first call, the time value of the first call is returned. One would have to call Calendar.clear() before each
- * get to obtain the current time and this class takes care of that for you.
- * 
- * Further, this class is synchronized so you do not need to worry about synchronization of a Calendar anymore.
+ * A helper class, which returns the current time and date in UTC timezone.
  */
 public class CurrentTimeUTC {
 
-	private static final GregorianCalendar mCalendar = new GregorianCalendar(TimeZone.getTimeZone("UTC"));
-
-	public static Date get() {
-		synchronized(mCalendar) {
-			mCalendar.setTimeInMillis(System.currentTimeMillis());
-			return mCalendar.getTime();
-		}
-	}
-
-	/**
-	 * Get the current time in milliseconds. 
-	 * 
-	 * In the current implementation, this just returns System.currentTimeMilis(). You should however use CurrenTimeUTC.getInMillis() instead because
-	 * the JavaDoc of System.currentTimeMilis() does not explicitly state what time zone it returns. Therefore, by using this wrapper function, your code
-	 * clearly states that it uses UTC time. 
-	 */
-	public static long getInMillis() {
-		return System.currentTimeMillis();
-	}
-
-	public static int getYear() {
-		synchronized(mCalendar) {
-			mCalendar.setTimeInMillis(System.currentTimeMillis());
-			return mCalendar.get(Calendar.YEAR);
-		}
-	}
-
-	public static int getMonth() {
-		synchronized(mCalendar) {
-			mCalendar.setTimeInMillis(System.currentTimeMillis());
-			return mCalendar.get(Calendar.MONTH);
-		}
-	}
-
-	public static int getDayOfMonth() {
-		synchronized(mCalendar) {
-			mCalendar.setTimeInMillis(System.currentTimeMillis());
-			return mCalendar.get(Calendar.DAY_OF_MONTH);
-		}
-	}
-
+    public static Date get() {
+        return Date.from(nowUtc().toInstant());
+    }
+
+    /**
+     * Get the current time in milliseconds.
+     *
+     * In the current implementation, this just returns {@link System#currentTimeMillis()}.
+     * You should however use CurrentTimeUTC#getInMillis() instead because
+     * the JavaDoc of {@link System#currentTimeMillis()} does not explicitly state what time zone it returns.
+     * Therefore, by using this wrapper function, your code clearly states that it uses UTC time.
+     */
+    public static long getInMillis() {
```

Comment:
Oh my, are we using this function anywhere? The documentation is horribly incorrect.

Reply:
Oof, the whole class is a serious clusterfuck. We should deprecate it right now (and nuke it from orbit, it’s the only way to be sure).

Reply:
@Bombe this sounds like “this is a good change and it would be good to do more changes” — am I right in that understanding of your comment?

Nuking the class from orbit sounds dangerous, because it is available to plugins (our usual problem). I like getting rid of unnecessary synchronization.

Reply:
At work we have the policy that every deprecation needs a comment people can copy paste to replace the usage of the deprecated method. I would suggest doing that here, too.

We cannot remove this until there isn’t a single use in plugins.

(besides: welcome back, @bertm !)

Reply:
Yeah, take a guess in which plugins this is used… we’re never going to get rid of this. 😭